pkg/controller/sparkapplication/controller.go

```diff
@@ -546,63 +545,52 @@ func (c *Controller) submitSparkApplication(app *v1alpha1.SparkApplication) *v1a
 				State:        v1alpha1.FailedSubmissionState,
 				ErrorMessage: err.Error(),
 			},
-			SubmissionAttempts:        app.Status.SubmissionAttempts + 1,
-			LastSubmissionAttemptTime: metav1.Now(),
 		}
 		return app
 	}
 
-	// Try submitting the application by running spark-submit.
-	submitted, err := runSparkSubmit(newSubmission(submissionCmdArgs, appToSubmit))
+	_, err = c.subJobManager.createSubmissionJob(newSubmission(submissionCmdArgs, appToSubmit))
 	if err != nil {
-		app.Status = v1alpha1.SparkApplicationStatus{
-			AppState: v1alpha1.ApplicationState{
-				State:        v1alpha1.FailedSubmissionState,
-				ErrorMessage: err.Error(),
-			},
-			SubmissionAttempts:        app.Status.SubmissionAttempts + 1,
-			LastSubmissionAttemptTime: metav1.Now(),
+		if !errors.IsAlreadyExists(err) {
+			app.Status = v1alpha1.SparkApplicationStatus{
+				AppState: v1alpha1.ApplicationState{
+					State:        v1alpha1.FailedSubmissionState,
```

Comment:
Looks like we are still setting the state to FailedSubmissionState at a couple of places in this method. We will need to handle retries for these errors when we fail without even creating a Job and we can't get rid of the logic in https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/358/files#diff-b7eea120d7ac0e8587781b863b7db8e4L479 . Sorry I missed this earlier in the last review.

Reply:
Restored.